cds_ils/migrator/items/utils.py

```diff
@@ -61,15 +61,25 @@ def clean_description_field(record):
         del record["description"]
 
 
-def clean_item_record(record):
+def clean_medium(record, document):
+    """Translates the item medium."""
+    if document["_migration"]["has_medium"]:
+        for item in document["_migration"]["item_medium"]:
+            if record["barcode"] == item["barcode"]:
+                record["medium"] = item["medium"]
+    else:
+        record["medium"] = "PAPER"  # requested as default value
+
```

Comment:
```suggestion
# this as another helper function, for finding documents, instead of adding it in the set_document_pid (in simplified version, we need the exceptions etc, like above)
def find_document_for_item(item_json):
    document = None
    try:
        document = get_record_by_legacy_recid(document_cls, item_json["id_bibrec"])
    except PIDDoesNotExistError as e:
        document = get_document_by_barcode(item_json["barcode"])
    if document is None:
        raise ....
    return document

def clean_medium(record):
    """Translates the item medium."""
    
    document = find_document_for_item(record)
  
    if document["_migration"]["has_medium"]:
        for item in document["_migration"]["item_medium"]:
            if record["barcode"] == item["barcode"]:
                record["medium"] = item["medium"]
    else:
        record["medium"] = "PAPER"  # requested as default value

```

Reply:
Hmm, I see, but this way we are calling the `find_document_for_item` function twice, this means that we will do for each item 2 searches that will retrieve the same document, when I kept it this way I had in mind avoiding this, is it OK to change the name of `set_document_pid` to `set_and_get_document` and retrieve the document or should I apply the suggested change? @kprzerwa

Reply:
true it will be two searches. But since in the first time we call the search we set the `document_pid` on the item. SO here we can simply do
```
document = document_class.get_record_by_pid(record["document_pid"])
```
and this one is not a search

Reply:
Ok, thanks!